### PR TITLE
[VCS] Use Task instead of threadpool for managing command running.

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlTask.cs [hunk 1]
```diff
@@ -42,21 +41,23 @@ protected virtual ProgressMonitor CreateProgressMonitor ()
...
+				tracker.ReportError (msg + e.Message, null);
+				LoggingService.LogError ("Version Control command failed: ", e);
 			} catch (Exception e) {
 				string msg = GettextCatalog.GetString ("Version control operation failed: ");
 				tracker.ReportError (msg, e);
-			} finally {			
-				threadnotify.WakeupMain();
+				LoggingService.LogError ("Version Control command failed: ", e);
+			} finally {
+				Wakeup ();
```
Comment (alanmcgovern) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlTask.cs]: This broke the threading semantics. Code which is required to run on the UI thread is now running on a background thread.

To retain the original semantics maybe it'd make sense to make the code something like...

```
async void BackgroundWorker ()
{
    await Task.Run (() => Run ());
    // the rest of the code is now on the UI thread and Wakeup will be invoked on the main thread.
}

Comment (Therzok) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/VersionControlTask.cs]: Yeah, I was not thinking straight when I wrote this, that's why it's on a PR.

**Comment by Therzok** [comment]
@alanmcgovern bump

**Comment by alanmcgovern** [comment]
LGTM!

**Comment by alanmcgovern** [comment]
Merged!

**Pull request closed without merging**